pkg/plugin/aggregation/run.go

```diff
@@ -150,18 +150,46 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	}()
 
 	// 4. Launch each plugin, to dispatch workers which submit the results back
-	for _, p := range plugins {
-		cert, err := auth.ClientKeyPair(p.GetName())
-		if err != nil {
-			return errors.Wrapf(err, "couldn't make certificate for plugin %v", p.GetName())
-		}
-		logrus.WithField("plugin", p.GetName()).Info("Running plugin")
-		if err = p.Run(client, cfg.AdvertiseAddress, cert); err != nil {
-			return errors.Wrapf(err, "error running plugin %v", p.GetName())
+	go func() {
+		pluginsRun := map[int]bool{}
+		for {
+			for index, p := range plugins {
+				// Plugin already run.
+				if pluginsRun[index] {
+					continue
+				}
+
+				// If unable to run plugin, wait.
+				if !canStartPlugin(aggr, p) {
+					logrus.Info("schnake: cant start plugin, continue")
+					continue
```

Comment:
Hi @johnSchnake,

This `continue` has no sleep, and neither does the loop's beginning. Thus if the plug-in is unable to start, and there's only one plug-in, it will immediately try again. Is this desired?

Reply:
👍 no, thanks for the find. When fixing up the PR I'll be sure to move the sleep (or something) around appropriately.